### PR TITLE
Add Mail::SPF perl module

### PR DESCRIPTION
**Pull request by fritzkink**
@mtelka please check why tests are not working. Also please check why *.p5m was not fully build correctly and whether my adaptions in perl-integrate-module.conf are reasonable.

### CODE REVIEW

**Comment by mtelka** [comment]
> @mtelka please check why tests are not working. Also please check why *.p5m was not fully build correctly and whether my adaptions in perl-integrate-module.conf are reasonable.

The first problem you have is license.  You need `%hook-no-license%` to solve that.

Since this module primarily supports `modulebuild` you should switch to `BUILD_STYLE = modulebuild`.  This should solve the problem with installation and sample-manifest.

For an example how to do that (all of the above) please look at `Encode-Detect/perl-integrate-module.conf`.

For failed tests you need to find the root cause of the failure and solve that.

There is also one additional problem: `usr/sbin/spfd`.  This file cannot be installed like that because you'll end up with two packages (-534 and -536) containing the same file.  Such packages would be impossible to install together.

**Comment by fritzkink** [comment]
Thanks should be ok now.

File: components/perl/Mail-SPF/perl-integrate-module.conf
Comment (mtelka) [inline components/perl/Mail-SPF/perl-integrate-module.conf]: I suggest to add something like:

```
GENERATE_EXTRA_CMD += | grep -v spfd
```

into `%include-3%`.  This is far simpler and it will also remove the file from sample manifest.

File: components/perl/Mail-SPF/perl-integrate-module.conf
Comment (mtelka) [inline components/perl/Mail-SPF/perl-integrate-module.conf]: Here it would be far better to add a patch that either fixes the test (the patch is attached to the ticket), or skips the test (a suggestion how to do that is in the ticket too).

File: components/perl/Mail-SPF/perl-integrate-module.conf
Comment (mtelka) [inline components/perl/Mail-SPF/perl-integrate-module.conf]: The license is definitely not MPL-1.1.  It looks like it is some kind of BSD, but you need to find which one: https://spdx.org/licenses/

**Review by mtelka — CHANGES_REQUESTED** [review]

File: components/perl/Mail-SPF/perl-integrate-module.conf
Comment (fritzkink) [reply components/perl/Mail-SPF/perl-integrate-module.conf]: No I don't, the author expressed his thoughts in a file called LICENSE which we should respect. Assuming something else without feedback from the author is simply wrong. I'll remove of course MPL-1.1 but the Mail-SPF.license file contains only the original license file.

File: components/perl/Mail-SPF/perl-integrate-module.conf
Comment (fritzkink) [reply components/perl/Mail-SPF/perl-integrate-module.conf]: Patching as suggested did not resolve completely the issue (different results between the perl versions). I'm not aware that openindiana provides distroprefs for perl where the mentioned file should land. Deleting the file simply also skips the test.

File: components/perl/Mail-SPF/perl-integrate-module.conf
Comment (fritzkink) [reply components/perl/Mail-SPF/perl-integrate-module.conf]: ok

File: components/perl/Mail-SPF/perl-integrate-module.conf
Comment (mtelka) [reply components/perl/Mail-SPF/perl-integrate-module.conf]: > No I don't, the author expressed his thoughts in a file called LICENSE which we should respect. Assuming something else without feedback from the author is simply wrong. I'll remove of course MPL-1.1 but the Mail-SPF.license file contains only the original license file.

Sorry, I do not understand.  Please elaborate.  Do you agree that the `LICENSE` file (and `Mail-SPF.license` file) contains some kind of BSD-style license?

Comment (fritzkink) [reply components/perl/Mail-SPF/perl-integrate-module.conf]: This section is part of almost any license terms. And "some kind of" is exactly what I mean with assumption.

File: components/perl/Mail-SPF/perl-integrate-module.conf
Comment (mtelka) [inline components/perl/Mail-SPF/perl-integrate-module.conf]: This could be safely dropped after #9377 is merged.

File: components/perl/Mail-SPF/perl-integrate-module.conf
Comment (mtelka) [reply components/perl/Mail-SPF/perl-integrate-module.conf]: > This section is part of almost any license terms. And "some kind of" is exactly what I mean with assumption.

What you refer to with "this section"?

Sorry, please elaborate about "assumption" more.  I'm afraid I still do not understand what you try to say by all of that.

And, I still miss answer to my question: Do you agree that the `LICENSE` file (and `Mail-SPF.license` file) contains some kind of BSD-style license?

If I know the answer to that question maybe I'd better understand what problem you see here and there.

Thanks.

Comment (fritzkink) [reply components/perl/Mail-SPF/perl-integrate-module.conf]: With "this section" I mean the stanza in capital letters in the license file. 
License is a strong legal term. If you change the license because you think it contains "some kind of" what ever other license without asking for feedback from the original author(s) you are liable for the consequences. Just a simple example:
If the license just contains "You can use the source for what ever you want but you are by any means not allowed to change the code" and you replace it because it contains "some kind of" whatever known license and this 15 page license contains a small stanza stating "in case of a bug you are allowed to fix it", you turn it into the opposite of what the author(s) intended.  And now, who is liable in case of a legal proceedings?

Basically it is not about whether I agree to your statement "contains some kind of BSD-style license". It is about we are not allowed to change whatever license.

File: components/perl/Mail-SPF/perl-integrate-module.conf
Comment (mtelka) [inline components/perl/Mail-SPF/perl-integrate-module.conf]: Please wait until #9451 is merged, then rebase on top of it, remove this line and re-run the `perl-integrate-module` script.  Thanks.

**Review by mtelka — CHANGES_REQUESTED** [review]

**Comment by AndWac** [comment]
@mtelka: is this ready for merge now?

File: components/perl/Mail-SPF/perl-integrate-module.conf
Comment (mtelka) [reply components/perl/Mail-SPF/perl-integrate-module.conf]: Please elaborate.  I just tried to run `perl-integrate-module` with this patch: https://rt.cpan.org/Ticket/Attachment/1603414/857647/Util.pm-patch.gz and `COMPONENT_PREP_ACTION` line removed from `perl-integrate-module.conf` and the run passed without any warning (or error).

Comment (mtelka) [reply components/perl/Mail-SPF/perl-integrate-module.conf]: If this line will stay here (I believe it shouldn't), please replace `rm -f` by `$(RM)`.

File: components/perl/Mail-SPF/perl-integrate-module.conf
Comment (mtelka) [inline components/perl/Mail-SPF/perl-integrate-module.conf]: Please replace `grep` by `$(GNU_GREP)`.

**Review by mtelka — CHANGES_REQUESTED** [review]

**Comment by fritzkink** [comment]
Are there any further comments/suggerstions?

File: components/perl/Mail-SPF/Makefile
Comment (AndWac) [inline components/perl/Mail-SPF/Makefile:23]: As we don't have any release of this package yet the COMPONENT_REVISION should be 0 or unset.

Comment (fritzkink) [reply components/perl/Mail-SPF/Makefile:23]: Sorry overseen. Fixed now.

**Review by AndWac — APPROVED** [review]

**Comment by mtelka** [comment]
> Are there any further comments/suggerstions?

You didn't respond yet to my comment https://github.com/OpenIndiana/oi-userland/pull/9358#discussion_r986140941.

**Comment by mtelka** [comment]
@fritzkink, please fix the following problem.  Thanks.

```
# pkg install -v library/perl-5/mail-spf
Creating Plan (Checking for conflicting actions): /           Packages to install:  3
       Create boot environment: No
Create backup boot environment: No
          Rebuild boot archive: No

Changed packages:
openindiana.org
  library/perl-5/mail-spf
    None -> 2.9.0-2022.0.0.0
  library/perl-5/mail-spf-534
    None -> 2.9.0-2022.0.0.0
  library/perl-5/mail-spf-536
    None -> 2.9.0-2022.0.0.0

pkg install: The following packages deliver conflicting action types to usr/bin/spfquery:

  file:
    pkg://openindiana.org/library/libspf2@1.2.11,5.11-2022.0.0.0:20221001T152656Z
  link:
    pkg://openindiana.org/library/perl-5/mail-spf-534@2.9.0,5.11-2022.0.0.0:20221030T185222Z
    pkg://openindiana.org/library/perl-5/mail-spf-536@2.9.0,5.11-2022.0.0.0:20221030T185225Z

These packages cannot be installed together. Any non-conflicting subset
of the above packages can be installed.
#
```

**Comment by fritzkink** [comment]
thank you for catching this conflict. New pull request uploaded.

**Pull request merged**
